src/makeInfinityAdder.js

```diff
@@ -4,7 +4,29 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let i = 0;
+
+  const adder = (valueFirst) => {
+    if (i === 0 && !valueFirst) {
+      sum = 0;
+
+      return sum;
+    }
+
+    if (valueFirst || valueFirst === 0) {
+      sum += valueFirst;
+      i++;
```

Comment:
The increment of 'i' is unnecessary and can be removed to simplify the function.